docs/remote_inference_blueprints/gcp_vertexai_connector_embedding_blueprint.md

```diff
@@ -0,0 +1,267 @@
+
+```
+# Authenticate using service account key
+gcloud auth activate-service-account --key-file=/path/to/service-account-key.json
+
+# Obtain access token
+gcloud auth print-access-token
+```
+
+## 3. Create model group:
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "remote_model_group_gcp",
+    "description": "This is an example group for embedding models"
+}
+```
+
+Sample response:
+```json
+{
+  "model_group_id": "BPtPEIwBqYi_Zeg-SR7R",
+  "status": "CREATED"
+}
+```
+
+
+## 3. Register model for VertexAI embedding model:
+
+Refer to [VertexAI Service REST API reference - Embedding](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings#get_text_embeddings_for_a_snippet_of_text).
+
+In order to use this, you need to supply the values for the below attributes
+
+* Project Id
+* Model Id
+* Access token
+
+
+```json
+POST /_plugins/_ml/models/_register
+{
+    "name": "vertexAI: model to generate embeddings",
+    "function_name": "remote",
+    "model_group_id": "BPtPEIwBqYi_Zeg-SR7R",
+    "description": "test vertexAI model",
+    "connector": {
+        "name": "VertexAI Connector",
+        "description": "The connector to public vertextAI model service for text embedding",
+        "version": 1,
+        "protocol": "http",
+        "parameters": {
+            "project": "<YOUR PROJECT_ID>",
+            "model_id": "<YOUR MODEL_ID>"
+        },
+        "credential": {
+            "vertexAI_token": "<YOUR ACCESS TOKEN>"
+        },
+        "actions": [
+            {
+                "action_type": "predict",
+                "method": "POST",
+                "url": "https://us-central1-aiplatform.googleapis.com/v1/projects/${parameters.project}/locations/us-central1/publishers/google/models/${parameters.model_id}:predict",
+                "headers": {
+                    "Authorization": "Bearer ${credential.vertexAI_token}"
+                },
+                "request_body": "{\"instances\": [{ \"content\": \"${parameters.prompt}\"}]}"
+            }
+        ]
+    }
+}
+```
+
+Sample response:
+```json
+{
+  "task_id": "pX8scY4B2QHLlv0i6LYZ",
+  "status": "CREATED",
+  "model_id": "pn8scY4B2QHLlv0i6LZB"
+}
+```
+
+## 4. Check the model registration
+```json
+GET /_plugins/_ml/tasks/pX8scY4B2QHLlv0i6LYZ
+```
+Sample response:
+
+
+```Response:
+{
+  "model_id": "pn8scY4B2QHLlv0i6LZB",
+  "task_type": "REGISTER_MODEL",
+  "function_name": "REMOTE",
+  "state": "COMPLETED",
+  "worker_node": [
+    "pQg-fbRpSm-x-V-0ICQpQw"
+  ],
+  "create_time": 1711295752216,
+  "last_update_time": 1711295752273,
+  "is_async": false
+}
+```
+
+## 5. Deploy the model
+
+```json
+POST /_plugins/_ml/models/pX8scY4B2QHLlv0i6LYZ/_deploy
+```
+Sample response:
+
+```Response:
+{
+  "task_id": "p38xcY4B2QHLlv0iSbb_",
+  "task_type": "DEPLOY_MODEL",
+  "status": "COMPLETED"
+}
+```
+
+## 6. Check model deployment
+```json
+GET /_plugins/_ml/tasks/pX8scY4B2QHLlv0i6LYZ
```

Comment:
this should be the deployment task id from above `p38xcY4B2QHLlv0iSbb_`

Reply:
Thanks @mingshl I missed it, I have corrected this now, please review